anvio/scgsdatabase.py

```diff
@@ -0,0 +1,791 @@
+
+
+### open fasta once.... make db of Bacteria_71 fasta .......
+class scgsdatabase():
+    def __init__(self, args, run=run, progress=progress):
+        self.args = args
+        self.run = run
+        self.progress = progress
+
+        A = lambda x: args.__dict__[x] if x in args.__dict__ else None
+
+        self.genes_files_directory = args.genes_files_directory
+
+        self.scgs_directory = args.scgs_directory
+
+        #self.num_threads=args.num_threads
+
+
+
+        self.dictionnary_level=self.dictionnary_level()
+
+
+
+        if not args.taxofiles:
+            self.taxofiles = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/SCG/merge_taxonomy.tsv')
+
+        #if not args.outtsv:
+        self.outtsv = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/SCG/mergedb/matching_taxonomy.tsv')
+
+        #if not args.num_threads:
+        self.num_threads=args.num_threads
+
+        if not args.num_threads:
+            self.num_threads=2
+
+
+        self.hmms=args.hmms
+
+        if not args.genes_files_directory:
+            self.genes_files_directory = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/SCG/msa_individual_genes')
+
+        self.genesfiles=os.listdir(self.genes_files_directory)
+
+
+        if not args.outputdirectory:
+            outputdirectory = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/SCG/mergedb')
+            if not os.path.exists(outputdirectory):
+                os.mkdir(os.path.dirname(anvio.__file__)+'/data/misc/SCG/mergedb')
+            self.outputdirectory = outputdirectory
+
+
+
+        self.scgsfasa = [files for files in os.listdir(
+            self.scgs_directory) if not files.endswith(".dmnd")]
+
+
+
+
+        dictionnary_corres={}
+
+        matrix_taxonomy=self.domatrix(self.taxofiles)
+
+        pathpickle_dictionnary__corres = os.path.join(
+            self.outputdirectory, 'dictionnary__corres.pickle')
+        if not os.path.exists(pathpickle_dictionnary__corres):
+            for gene in self.genesfiles:
+                #for dbdiamond
+                name=gene.replace('.faa','')
+                path_diamondb=os.path.join(self.outputdirectory, "diamonddb")
+                pathdb=os.path.join(self.genes_files_directory, name)
+                pathgenes=os.path.join(self.genes_files_directory, gene)
+                outpath=os.path.join(self.outputdirectory, "diamondblast")
+                pathrefundgenes=os.path.join(path_diamondb, gene)
+                pathblast=os.path.join(outpath,name)
+                if not os.path.exists(path_diamondb):
+                    os.mkdir(path_diamondb)
+
+
+                if not os.path.exists(outpath):
+                    os.mkdir(outpath)
+                    outpath=os.path.join(self.outputdirectory, "diamondblast")
+
+
+                sequencetoblast=self.refundgenes(pathgenes,pathrefundgenes,pathdb)
+                self.diamonddb(self.hmms)
+                if not os.path.exists(pathrefundgenes):
```

Comment:
again, variable names like `pathrefundgenes` make no sense :( take a look at this function, for instance:

https://github.com/merenlab/anvio/blob/master/anvio/dbops.py#L364

don't worry about having long variable names. it is much better to have long names than having cryptic names.